cli/flox-rust-sdk/src/flox.rs

```diff
@@ -192,45 +192,80 @@ pub enum FloxhubError {
     InvalidFloxhubBaseUrl(String, #[source] url::ParseError),
 }
 
-use tempfile::TempDir;
-/// Should only be used in the flox crate
-pub fn test_flox_instance() -> (Flox, TempDir) {
-    use crate::models::environment::{global_manifest_path, init_global_manifest};
-
-    let tempdir_handle = tempfile::tempdir_in(std::env::temp_dir()).unwrap();
-
-    let cache_dir = tempdir_handle.path().join("caches");
-    let data_dir = tempdir_handle.path().join(".local/share/flox");
-    let temp_dir = tempdir_handle.path().join("temp");
-    let config_dir = tempdir_handle.path().join("config");
-
-    std::fs::create_dir_all(&cache_dir).unwrap();
-    std::fs::create_dir_all(&temp_dir).unwrap();
-    std::fs::create_dir_all(&config_dir).unwrap();
-
-    let flox = Flox {
-        system: env!("NIX_TARGET_SYSTEM").to_string(),
-        cache_dir,
-        data_dir,
-        temp_dir,
-        config_dir,
-        access_tokens: Default::default(),
-        netrc_file: Default::default(),
-        uuid: Default::default(),
-        floxhub: Floxhub::new(Url::from_str("https://hub.flox.dev").unwrap(), None).unwrap(),
-        floxhub_token: None,
-    };
-
-    init_global_manifest(&global_manifest_path(&flox)).unwrap();
-
-    (flox, tempdir_handle)
+pub mod test_helpers {
+    use tempfile::TempDir;
+
+    use super::*;
+    use crate::models::environment::global_manifest_lockfile_path;
+    use crate::models::lockfile::LockedManifest;
+
+    /// Get an instance of Flox that has a locked global manifest.
+    ///
+    /// This means any operations that use pkgdb should use the same nixpkgs
+    /// revision.
+    pub fn flox_instance_with_global_lock() -> (Flox, TempDir) {
+        // Scrape nixpkgs once and then store the resulting global lockfile in memory
+        static GLOBAL_LOCKFILE: Lazy<LockedManifest> = Lazy::new(|| {
+            let (flox, _temp_dir_handle) = flox_instance();
+            let pkgdb_nixpkgs_rev_new = "ab5fd150146dcfe41fda501134e6503932cc8dfd";
```

Comment:
Nit: we should probably reduce the number of places we have to list the literal rev. Not sure this PR is the place to do it. I remember when I was working on the fetcher it was annoying to have to go update the NAR hashes in several different places every time I modified something.

Reply:
Yeah not ideal. I think once per language is probably acceptable cause it's easy to grep for though. We could set it in default.nix and that would be pretty easy to use in Rust, but I don't know how we'd get that into stuff like all of our JSON test files. I'll skip addressing that in this PR